core/deployment/src/main/java/io/quarkus/deployment/ExtensionLoader.java

```diff
@@ -432,6 +433,7 @@ private static Consumer<BuildChainBuilder> loadStepsFromClass(Class<?> clazz,
         final List<Method> methods = getMethods(clazz);
         final Map<String, List<Method>> nameToMethods = methods.stream().collect(Collectors.groupingBy(m -> m.getName()));
 
+        MethodHandles.Lookup lookup = MethodHandles.lookup();
```

Comment:
Since we use `setAccessible` below when the class is not accessible, I'd just use the `publicLookup()` here instead. This will bypass annoying restrictions about what module reads what other module or what packages are exported or whatever.

Reply:
Makes sense!